pexec/managed_process_test.go

```diff
@@ -193,6 +209,63 @@ func TestManagedProcessManage(t *testing.T) {
 			test.That(t, err.Error(), test.ShouldContainSubstring, "exit status 1")
 		}
 	})
+	t.Run("OnCrashHandler", func(t *testing.T) {
+		t.Run("returns true and process is restarted", func(t *testing.T) {
+			logger := golog.NewTestLogger(t)
+
+			tempFile := testutils.TempFile(t, "something.txt")
+			defer tempFile.Close()
+
+			watcher, err := fsnotify.NewWatcher()
+			test.That(t, err, test.ShouldBeNil)
+			defer watcher.Close()
+			watcher.Add(tempFile.Name())
+
+			var onCrashHandlerCallCount int
+			proc := NewManagedProcess(ProcessConfig{
+				Name:           "bash",
+				Args:           []string{"-c", fmt.Sprintf("echo hello >> '%s'\nexit 1", tempFile.Name())},
+				OnCrashHandler: func() bool { onCrashHandlerCallCount++; return true },
```

Comment:
What stops this from just restarting indefinitely? Shouldn't it return false after, say, a count of 5 or something?

Reply:
To be clear, it _looks_ like it's a race condition... the watcher gets three hits and the call count is checked RIGHT after that, even though the process would continue to restart until proc.Stop(). If that's NOT the case, then please explain where I'm wrong.